icu4c/source/test/intltest/tzfmttst.cpp

```diff
@@ -1302,4 +1303,719 @@ TimeZoneFormatTest::TestFormatTZDBNamesAllZoneCoverage(void) {
     }
 }
 
+void
+TimeZoneFormatTest::TestLocaleTZExtension(void) {
+    UDate now = Calendar::getNow();
+
+    static const char* const tz[] = {
```

Comment:
same comments as in caltest.cpp -- number of test cases, dataerrln vs. assertEquals, not UTF-8, ...